.github/workflows/artillery.yml

```
@@ -0,0 +1,30 @@
+name: Load Test
+
+on:
+  push:
+    branches:
+      - main  
+  pull_request:
+    branches:
+      - main  
+
+jobs:
+  load-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Run load test with Artillery
+        uses: artilleryio/action-cli@v1
+        env:
+          ARTILLERY_AUTH_TOKEN: ${{ secrets.ARTILLERY_AUTH_TOKEN }}
+        with:
+          command: "run --output test-report.json load-test.yml"
+
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: artillery-test-results
+          path: test-report.json
```

I suggest you put this report file inside a folder for better clarity.
```suggestion
          path: reports/report.json
```